Node Hog Scenario failures during container execution

## Problem

During one of test runs in https://github.com/krkn-chaos/krkn-ai/pull/65, it was observed that while running node related scenarios there is a consistent failure caused by HTTP request error (check logs below).

To reproduce the issue:
1. Create `krkn-ai.yaml` config either via local cli or podman.
2. Keep only the `node-hog` scenarios and disable rest.
3. Run the updated `krkn-ai.yaml` config in podman container:
     ```
     podman run --rm \
     --privileged \
     -v ./tmp/container:/mount:Z \
     -e MODE=run \
     -e CONFIG_FILE="/mount/krkn-ai.yaml" \
     -e KUBECONFIG="/mount/kubeconfig.yaml" \
     -e OUTPUT_DIR="/mount/result/" \
     -e EXTRA_PARAMS="HOST=${HOST}" \
     -e VERBOSE=2 \
     krkn-ai:latest
     ```

This will result in failure, while if we try to run the same krkn-ai.yaml without container then it works without any issue.

## Scenario
```yaml
generation_id: 0
scenario_id: 1
scenario:
  name: node-memory-hog
  krknctl_name: node-memory-hog
  krknhub_image: containers.krkn-chaos.dev/krkn-chaos/krkn-hub:node-memory-hog
cmd: podman run --env-host=true -e PUBLISH_KRAKEN_STATUS="False" -e TELEMETRY_PROMETHEUS_BACKUP="False"
  -e WAIT_DURATION=0  -e TOTAL_CHAOS_DURATION="60"  -e MEMORY_CONSUMPTION_PERCENTAGE="42%"  -e
  NUMBER_OF_WORKERS="1"  -e NODE_SELECTOR="kubernetes.io/hostname=ip-10-0-1-58.ec2.internal"  -e
  TAINTS="[]"  -e NUMBER_OF_NODES="1"  -e IMAGE="quay.io/krkn-chaos/krkn-hog"  --net=host
  -v /mount/kubeconfig.yaml:/home/krkn/.kube/config:Z containers.krkn-chaos.dev/krkn-chaos/krkn-hub:node-memory-hog
log: /mount/result/logs/scenario_1.log
returncode: 1
start_time: '2025-12-11T11:59:28.032947'
end_time: '2025-12-11T12:01:10.082994'
fitness_result:
  scores: []
  health_check_failure_score: 0.0
  health_check_response_time_score: 0.0
  krkn_failure_score: -1.0
  fitness_score: -1.0
health_check_results:
...
...
```

## Error Log
```log
[1m11-12-2025T11:59:28 Checking if OpenShift client is installed[0m
/usr/bin/oc
2025-12-11 11:59:29,398 [INFO] Starting kraken
2025-12-11 11:59:29,405 [INFO] Initializing client to talk to the Kubernetes cluster
2025-12-11 11:59:29,405 [INFO] Generated a uuid for the run: 0199ec34-3c55-4a00-bc2c-8d0e75b8dafe
2025-12-11 11:59:30,134 [INFO] Detected distribution openshift
2025-12-11 11:59:36,827 [INFO] Fetching cluster info
2025-12-11 11:59:38,971 [INFO] 4.19.21
2025-12-11 11:59:38,972 [INFO] Server URL: https://api.j6yws-6s5o6-b2m.dsiy.p3.openshiftapps.com:443
2025-12-11 11:59:38,972 [INFO] Daemon mode not enabled, will run through 1 iterations

/usr/local/lib/python3.9/site-packages/google/api_core/_python_version_support.py:252: FutureWarning: You are using a Python version (3.9.21) past its end of life. Google will update google.api_core with critical bug fixes on a best-effort basis, but not with any other fixes or features. Please upgrade to the latest Python version, or at least Python 3.10, and then update google.api_core.
  warnings.warn(message, FutureWarning)
2025-12-11 11:59:40,297 [INFO] 📣 `ScenarioPluginFactory`: types from config.yaml mapped to respective classes for execution:
2025-12-11 11:59:40,298 [INFO]   ✅ type: application_outages_scenarios ➡️ `ApplicationOutageScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: container_scenarios ➡️ `ContainerScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: hog_scenarios ➡️ `HogsScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: kubevirt_vm_outage ➡️ `KubevirtVmOutageScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: managedcluster_scenarios ➡️ `ManagedClusterScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ types: [pod_network_scenarios, ingress_node_scenarios] ➡️ `NativeScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: network_chaos_scenarios ➡️ `NetworkChaosScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: network_chaos_ng_scenarios ➡️ `NetworkChaosNgScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: node_scenarios ➡️ `NodeActionsScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: pod_disruption_scenarios ➡️ `PodDisruptionScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: pvc_scenarios ➡️ `PvcScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: service_disruption_scenarios ➡️ `ServiceDisruptionScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: service_hijacking_scenarios ➡️ `ServiceHijackingScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: cluster_shut_down_scenarios ➡️ `ShutDownScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: syn_flood_scenarios ➡️ `SynFloodScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: time_scenarios ➡️ `TimeActionsScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO]   ✅ type: zone_outages_scenarios ➡️ `ZoneOutageScenarioPlugin` 
2025-12-11 11:59:40,298 [INFO] 

2025-12-11 11:59:40,299 [INFO] health checks config is not defined, skipping them
2025-12-11 11:59:40,299 [INFO] kube virt checks config is not defined, skipping them
2025-12-11 11:59:40,299 [INFO] Executing scenarios for iteration 0
2025-12-11 11:59:40,301 [INFO] Signal handlers registered globally
2025-12-11 11:59:40,301 [INFO] Running HogsScenarioPlugin: ['hog_scenarios'] -> scenarios/kube/memory-hog.yml
2025-12-11 11:59:40,301 [INFO] Set rollback_context: 1765454380301738448-0199ec34-3c55-4a00-bc2c-8d0e75b8dafe for scenario_type: hog_scenarios RollbackHandler
2025-12-11 11:59:42,177 [INFO] running memory hog scenario
2025-12-11 11:59:42,177 [INFO] targeting nodes: [ip-10-0-1-58.ec2.internal]
2025-12-11 11:59:42,177 [INFO] [ip-10-0-1-58.ec2.internal] workers number: 1
2025-12-11 11:59:44,427 [INFO] Serialized callable written to /tmp/kraken-rollback/1765454380301738448-0199ec34-3c55-4a00-bc2c-8d0e75b8dafe/hog_scenarios_1765454369386853753_vbrexlej.py
2025-12-11 11:59:44,427 [INFO] Rollback callable serialized to /tmp/kraken-rollback/1765454380301738448-0199ec34-3c55-4a00-bc2c-8d0e75b8dafe/hog_scenarios_1765454369386853753_vbrexlej.py
2025-12-11 11:59:45,218 [ERROR] Pod creation failed (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '4d6d9797-f9cb-4473-a098-5485093e4b1c', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains; preload', 'Warning': '299 - "unknown field \\"spec.imagePullPolicy\\""', 'X-Kubernetes-Pf-Flowschema-Uid': 'd2d0b35e-2987-41f5-ac5e-ef3d48055478', 'X-Kubernetes-Pf-Prioritylevel-Uid': '1859628d-1dd2-4c78-b71c-cff9b878c92e', 'Date': 'Thu, 11 Dec 2025 11:59:45 GMT', 'Content-Length': '184'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"namespaces \".*\" not found","reason":"NotFound","details":{"name":".*","kind":"namespaces"},"code":404}


2025-12-11 11:59:45,906 [ERROR] scenario exception: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '4d6d9797-f9cb-4473-a098-5485093e4b1c', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains; preload', 'Warning': '299 - "unknown field \\"spec.imagePullPolicy\\""', 'X-Kubernetes-Pf-Flowschema-Uid': 'd2d0b35e-2987-41f5-ac5e-ef3d48055478', 'X-Kubernetes-Pf-Prioritylevel-Uid': '1859628d-1dd2-4c78-b71c-cff9b878c92e', 'Date': 'Thu, 11 Dec 2025 11:59:45 GMT', 'Content-Length': '184'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"namespaces \".*\" not found","reason":"NotFound","details":{"name":".*","kind":"namespaces"},"code":404}


2025-12-11 11:59:45,907 [WARNING] Auto rollback is disabled, skipping execution for run_uuid=0199ec34-3c55-4a00-bc2c-8d0e75b8dafe, scenario_type=hog_scenarios
2025-12-11 11:59:46,639 [INFO] Find cluster events in file /tmp/events.json
2025-12-11 11:59:46,639 [INFO] wating 0 before running the next scenario
2025-12-11 11:59:46,640 [INFO] collecting OCP cluster metadata, this may take few minutes....
2025-12-11 12:01:07,304 [INFO] failed to parse virtualmachines API
2025-12-11 12:01:07,312 [INFO] Chaos data:
{
    "telemetry": {
        "scenarios": [
            {
                "start_timestamp": 1765454380,
                "end_timestamp": 1765454385,
                "scenario": "scenarios/kube/memory-hog.yml",
                "scenario_type": "hog_scenarios",
                "exit_status": 1,
                "parameters_base64": "",
                "parameters": {
                    "duration": 60,
                    "hog-type": "memory",
                    "image": "quay.io/krkn-chaos/krkn-hog",
                    "namespace": ".*",
                    "node-selector": "kubernetes.io/hostname=ip-10-0-1-58.ec2.internal",
                    "number-of-nodes": 1,
                    "taints": [],
                    "vm-bytes": "42%",
                    "workers": 1
                },
                "affected_pods": {
                    "recovered": [],
                    "unrecovered": [],
                    "error": null
                },
                "affected_nodes": [],
                "cluster_events": []
            }
        ],
        "node_summary_infos": [
            {
                "count": 2,
                "architecture": "amd64",
                "instance_type": "m5.xlarge",
                "nodes_type": "worker",
                "kernel_version": "5.14.0-570.72.1.el9_6.x86_64",
                "kubelet_version": "v1.32.10",
                "os_version": "Red Hat Enterprise Linux CoreOS 9.6.20251203-0 (Plow)"
            }
        ],
        "node_taints": [],
        "kubernetes_objects_count": {
            "ConfigMap": 276,
            "Pod": 94,
            "Deployment": 36,
            "Route": 9,
            "Build": 1
        },
        "network_plugins": [
            "OVNKubernetes"
        ],
        "timestamp": "2025-12-11T11:59:38Z",
        "health_checks": null,
        "virt_checks": null,
        "total_node_count": 2,
        "cloud_infrastructure": "AWS",
        "cloud_type": "rosa",
        "cluster_version": "4.19.21",
        "major_version": "4.19",
        "run_uuid": "0199ec34-3c55-4a00-bc2c-8d0e75b8dafe",
        "post_virt_checks": null,
        "job_status": false,
        "build_url": "manual"
    },
    "critical_alerts": null
}
2025-12-11 12:01:07,313 [INFO] telemetry collection disabled, skipping.
2025-12-11 12:01:07,313 [ERROR] Post scenarios are still failing at the end of all iterations
 _              _              
| | ___ __ __ _| | _____ _ __  
| |/ / '__/ _` | |/ / _ \ '_ \ 
|   <| | | (_| |   <  __/ | | |
|_|\_\_|  \__,_|_|\_\___|_| |_|
                               

An error occurred: module 'importlib.metadata' has no attribute 'packages_distributions'

```

--- a/krkn_ai/models/scenario/scenario_cpu_hog.py
+++ b/krkn_ai/models/scenario/scenario_cpu_hog.py
@@ -15,6 +15,7 @@ class NodeCPUHogScenario(Scenario):
     chaos_duration: TotalChaosDurationParameter = TotalChaosDurationParameter()
     # node_cpu_core: NodeCPUCoreParameter = NodeCPUCoreParameter()
     node_cpu_percentage: NodeCPUPercentageParameter = NodeCPUPercentageParameter()
+    namespace: NamespaceParameter = NamespaceParameter(value="default")
     node_selector: NodeSelectorParameter = NodeSelectorParameter()
     taint: TaintParameter = TaintParameter()
     number_of_nodes: NumberOfNodesParameter = NumberOfNodesParameter()
@@ -30,6 +31,7 @@ class NodeCPUHogScenario(Scenario):
             self.chaos_duration,
             # self.node_cpu_core,
             self.node_cpu_percentage,
+            self.namespace,
             self.node_selector,
             self.taint,
             self.number_of_nodes,

--- a/krkn_ai/models/scenario/scenario_memory_hog.py
+++ b/krkn_ai/models/scenario/scenario_memory_hog.py
@@ -17,6 +17,7 @@ class NodeMemoryHogScenario(Scenario):
     chaos_duration: TotalChaosDurationParameter = TotalChaosDurationParameter()
     node_memory_percentage: NodeMemoryPercentageParameter = NodeMemoryPercentageParameter()
     number_of_workers: NumberOfWorkersParameter = NumberOfWorkersParameter()
+    namespace: NamespaceParameter = NamespaceParameter(value="default")
     node_selector: NodeSelectorParameter = NodeSelectorParameter()
     taint: TaintParameter = TaintParameter()
     number_of_nodes: NumberOfNodesParameter = NumberOfNodesParameter()
@@ -32,6 +33,7 @@ class NodeMemoryHogScenario(Scenario):
             self.chaos_duration,
             self.node_memory_percentage,
             self.number_of_workers,
+            self.namespace,
             self.node_selector,
             self.taint,
             self.number_of_nodes,

